lib/trino-record-decoder/src/main/java/io/trino/decoder/avro/AvroColumnDecoder.java

```diff
@@ -185,6 +192,12 @@ public boolean getBoolean()
         public long getLong()
         {
             if (value instanceof Long || value instanceof Integer) {
+                if (columnType == TimestampType.TIMESTAMP_MILLIS) {
+                    return ((Number) value).longValue() * Timestamps.MICROSECONDS_PER_MILLISECOND;
```

Comment:
Won't Avro reader perform the conversion from long and gives us the result in `LocalDateTime` instead of long ?

Reply:
Could you show me a code snippet for this statement?

I'd say from the tests it's spotted that the object here is `long` instead of `LocalDateTime`. Avro's timestamp has the type `long` with an annotated logical type `timestamp_miils`.

This holds after I update the type passed with logicalType part.